sycl/source/detail/scheduler/commands.cpp

```diff
@@ -169,12 +169,26 @@ static std::string commandToName(Command::CommandType Type) {
 }
 #endif
 
-static std::vector<RT::PiEvent>
-getPiEvents(const std::vector<EventImplPtr> &EventImpls) {
+std::vector<RT::PiEvent>
+Command::getPiEvents(const std::vector<EventImplPtr> &EventImpls) const {
   std::vector<RT::PiEvent> RetPiEvents;
   for (auto &EventImpl : EventImpls) {
-    if (EventImpl->getHandleRef() != nullptr)
-      RetPiEvents.push_back(EventImpl->getHandleRef());
+    if (EventImpl->getHandleRef() == nullptr)
```

Comment:
Won't `getHandleRef` return `nullptr` for host tasks?

Reply:
this function is used for MPreparedDepsEvents only. host tasks and MPreparedHostDepsEvents are rendered separately

Reply:
Does it mean that we cannot have an event for host task here?

Reply:
So, how then the check below(on line 187)
`(static_cast<const ExecCGCommand *>(this))->getCG().getType() != CG::CGTYPE::CodeplayHostTask))`
can pass?

Reply:
EventsImpls that we are rendering in getPIEvents function are dependencies for "this" command and are commands isPiExpected for which is true. The check on line 187 analyzes "this" command, not dependencies. We do this check to verify that we do not have switch between sync methods here (host task event sync for in order queue vs native sync in backend).
To conclude:
"this" here can be both host task or task which produces pi event.
EventImpls here are MPreparedDepsEvents for "this" and consist only of command that produces pi events.